### ISSUE TITLE
Add ability to edit token approve amount

### ISSUE THREAD

**Issue by nightzeroeth** [rainbow-me/browser-extension#1418]
To ensure safety, it's best to set the token approve amount to match the transaction amount by default. 

Alternatively, we could incorporate a feature allowing users to adjust the token approve amount as needed.

See the example below:

<img width="348" alt="Screenshot 2024-03-21 at 14 57 59" src="https://github.com/rainbow-me/browser-extension/assets/134239349/f8d7ee2a-d9b1-4259-b649-1c398b55221a">

**Comment by nightzeroeth** [comment]
I'm sure this will save a lot of wallets from being drained.

**Comment by Ncookiez** [comment]
> it's best to set the token approve amount to match the transaction amount by default

Absolutely. But the transaction request is made by the dapp you're interacting with, not your wallet. You should give them this feedback!

**Comment by nightzeroeth** [comment]
Then I guess it's better to add an edit button, so we don't just approve whatever amount the dapp is asking.
